domain/schema_validator.go

```diff
@@ -209,7 +220,35 @@ func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedTableIDs [
 
 func (s *schemaValidator) enqueue(schemaVersion int64, relatedTableIDs []int64) {
 	s.deltaSchemaInfos = append(s.deltaSchemaInfos, deltaSchemaInfo{schemaVersion, relatedTableIDs})
-	if len(s.deltaSchemaInfos) > maxNumberOfDiffsToLoad {
+	s.notMergeCnt++
+
+	maxCnt := int(variable.GetMaxDetalSchemaCount())
+	if len(s.deltaSchemaInfos) > maxCnt && s.notMergeCnt > maxCnt/2 {
+		s.merge()
+		s.notMergeCnt = 1
+	}
+	if len(s.deltaSchemaInfos) > maxCnt {
 		s.deltaSchemaInfos = s.deltaSchemaInfos[1:]
 	}
 }
+
+func equal(a, b []int64) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := 0; i < len(a); i++ {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (s *schemaValidator) merge() {
```

Comment:
Is it possible to try to merge delta info just at the first time of `enque`?
```
s.deltaSchemaInfos = append(s.deltaSchemaInfos, deltaSchemaInfo{schemaVersion, relatedTableIDs})
```

Reply:
Good idea. Done